traits/traits_listener.py

```diff
@@ -170,57 +184,81 @@ def handle_dict_items(self, object, name, old, new):
-    notify = Bool(True)
-
-    #: Should registering listeners for items reachable from this listener item
-    #: be deferred until the associated trait is first read or set?
-    deferred = Bool(False)
-
-    #: Is this an 'any_trait' change listener, or does it create explicit
-    #: listeners for each individual trait?
-    is_any_trait = Bool(False)
-
-    #: Is the associated handler a special list handler that handles both
-    #: 'foo' and 'foo_items' events by receiving a list of 'deleted' and
-    #: 'added' items as the 'old' and 'new' arguments?
-    is_list_handler = Bool(False)
+    Parameters
+    ----------
+    name : str
+        The name of the trait to listen to.
+    metadata_name : str, optional
+        The name of any metadata that must be present (or not present).
+    metadata_defined : bool, optional
+        True if the specified metadata needs to be defined; False if the
+        specified metadata needs to be not defined.
+    handler : ListenerHandler, optional
+        Zero-argument callable that returns the actual handler when called (or
+        Undefined if that handler is no longer available).
+    wrapped_handler_ref : weakref.ref, optional
+        Weak reference to a ListenerNotifyWrapper wrapping the actual handler.
+    dispatch : str
+        The dispatch mechanism to use when invoking the handler.
+    priority : bool, optional
+        True if the handler goes at the beginning of the notification handlers
+        list, else False.
+    next : ListenerBase or None, optional
+        The next level (if any) of ListenerBase object to be called when any
+        of this object's listened-to traits is changed. The default is None.
+    type : int
+        The type of handler being used. One of ANY_LISTENER, SRC_LISTENER,
+        DST_LISTENER.
+    notify : bool, optional
+        True if changes to this item should generate a notification to
+        the handler; False otherwise. The default is True.
+    deferred : bool, optional
+        True if registering listeners for items reachable from this listener
+        item should be deferred until the associated trait is first read or
+        set.
+    is_any_trait : bool, optional
+        True if this is an "anytrait" change listener. False if it creates
+        explicit listeners for each individual trait.
+    is_list_handler : bool, optional
+        True if the associated handler is a special list handler that handles
+        both "foo" and "foo_items" events. False otherwise.
+    """
 
-    #: A dictionary mapping objects to a list of all current active
-    #: (*name*, *type*) listener pairs, where *type* defines the type of
-    #: listener, one of: (SIMPLE_LISTENER, LIST_LISTENER, DICT_LISTENER).
-    active = Instance(WeakIDKeyDict, ())
+    def __init__(
+        self,
+        *,
+        name,
+        metadata_name="",
+        metadata_defined=True,
+        handler=None,
```

Comment:
`handler` and `wrapped_handler_ref` should be required, not optional, but there are currently some unit tests that don't pass them.  We can make these required in another PR that updates the test code accordingly.